blueocean-rest/src/main/java/io/jenkins/blueocean/analytics/Analytics.java

```diff
@@ -62,5 +68,49 @@ public static boolean isAnalyticsEnabled() {
     /**
      * @param req to track
      */
-    public abstract void track(TrackRequest req);
+    public final void track(TrackRequest req) {
+        if (req == null) {
+            throw new ServiceException.BadRequestException("missing request");
+        }
+        if (StringUtils.isEmpty(req.name)) {
+            throw new ServiceException.BadRequestException("missing name");
+        }
+        Map<String, Object> allProps = req.properties == null ? Maps.<String, Object>newHashMap() : Maps.newHashMap(req.properties);
+        // Enhance with additional properties
+        for (AdditionalAnalyticsProperties enhancer : ExtensionList.lookup(AdditionalAnalyticsProperties.class)) {
+            Map<String, Object> additionalProperties = enhancer.properties(req);
+            if (additionalProperties != null) {
+                allProps.putAll(additionalProperties);
+            }
+        }
+        allProps.put("jenkins", server());
+        allProps.put("userId", identity());
+        Objects.ToStringHelper eventHelper = Objects.toStringHelper(this).add("name", req.name).add("props", allProps);
+        try {
+            doTrack(req.name, allProps);
+            if (LOGGER.isLoggable(Level.FINE)) {
+                String msg = eventHelper.toString();
+                LOGGER.log(Level.FINE, msg);
+            }
+        } catch (Throwable throwable) {
+            String msg = eventHelper.toString();
+            LOGGER.log(Level.WARNING, "Failed to send event: " + msg);
+        }
+    }
+
+    /**
+     * @param name of the event
+     * @param allProps properties to be reported
+     */
+    protected abstract void doTrack(String name, Map<String, Object> allProps);
```

Comment:
I think just `public abstract void track(TrackRequest req);` was cleaner and better abstraction.